src/types.ts

```diff
@@ -1867,8 +1867,10 @@ function mergeValues(
   if (a === b) {
     return { valid: true, data: a };
   } else if (aType === ZodParsedType.object && bType === ZodParsedType.object) {
-    const bKeysSet = new Set(Object.keys(b));
-    const sharedKeys = Object.keys(a).filter((key) => bKeysSet.has(key));
+    const bKeys = util.objectKeys(b);
```

Comment:
Do not use Set for intersection as it is not supported in legacy mobile browsers (like Android 4.4.4).